File: 프로그래머스/0/181905. 문자열 뒤집기/문자열 뒤집기.py
def solution(my_string, s, e):
    # 1. 부분 문자열 추출
    substring = my_string[s:e+1]
    
    # 2. 부분 문자열 뒤집기
    reversed_substring = substring[::-1]
    
    # 3. 문자열 합치기
    result = my_string[:s] + reversed_substring + my_string[e+1:]
    
    return result

File: 프로그래머스/0/120822. 문자열 뒤집기/문자열 뒤집기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(const char* my_string) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int len=strlen(my_string);
    char* answer = (char*)malloc(len);
    for(int i=0;i<len;i++){
        answer[i]=my_string[len-i-1];
    }
    answer[len]=NULL;
    return answer;
}